cli/smartcontract/smart_contract.go

```diff
@@ -399,8 +399,10 @@ func invokeInternal(ctx *cli.Context, signAndPush bool) error {
 	if !args.Present() {
 		return cli.NewExitError(errNoScriptHash, 1)
 	}
-	script := args[0]
-
+	script, err := util.Uint160DecodeStringLE(args[0])
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("incorrect script hash: %v", err), 1)
+	}
 	if len(args) <= 1 {
 		return cli.NewExitError(errNoMethod, 1)
 	}
```

Comment:
As far as we're here, is it really required to pass `method` argument? I tried to invoke basic `HelloWorld` contract, which does not accept any `method` parameter and doesn't have it's own methods. However, cli does not allow to invoke smart contract without method, so specifying any random string instead of `method` makes it possible to invoke the contract.

Reply:
>  is it really required to pass method argument?

I think so. There is an `InvokeScript` for arbitrary script, but `InvokeFunction` is supposed to generate something that calls the method specified. It can be `Main`. Or something else. BTW, we need to check if our `InvokeFunction` is in fact correct, I'm not sure about that at the moment.